pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -448,14 +494,39 @@ private void commitSegmentMetadataInternal(String realtimeTableName,
     // Refresh the Broker routing to reflect the changes in the segment ZK metadata
     _helixResourceManager.sendSegmentRefreshMessage(realtimeTableName, committingSegmentName, false, true);
 
-    // Step-2
+    // Get current partition groups - this gives current state of latest segments for each partition
+    // E.g. [A - DONE], [B - IN_PROGRESS], [C - IN_PROGRESS]
+    PartitionLevelStreamConfig streamConfig = new PartitionLevelStreamConfig(tableConfig.getTableName(),
+        IngestionConfigUtils.getStreamConfigMap(tableConfig));
+    List<PartitionGroupMetadata> currentPartitionGroupMetadataList =
+        getCurrentPartitionGroupMetadataList(idealState, streamConfig);
+
+    // Fetches new partition groups, given current partition groups metadata.
+    // Assume stream has partitions A, B, C, all still consuming. Result will be A, B, C
+    // Assume A was split into D, E, but messages of A are yet to be consumed, result will be A, B, C
```

Comment:
This code runs in the controller. How does the controller know whether servers have consumed messages of partition A or not? There may be multiple replicas of servers, some servers have have consumed it, the others may not have. How does that reflect here?

Let me know if I am missing something, or is it just comments that need to change

Reply:
The comments are kinda irrelevant out here, as those are kinesis specific examples. Removed them.
In general, the controller will get the currentPartitionGroupMetadata using the ideal state and segment zk metadata of latest segment for each partition group. 
The Stream impl will look at the partitions in the stream and get new partition groups.
Take the example of Kafka, if we began with 3 partitions, and partition 0 was committing, we would have 
Current partition group metadata:
0 - seqNum: 0, startOffset: 100, endOffset: 220,  status: DONE
1 - seqNum: 0, startOffset: 110, endOffset: -,  status: IN_PROGRESS
2 - seqNum: 0, startOffset: 100, endOffset: -,  status: IN_PROGRESS
And new partition group info: [0, 1, 2]

Lets assume partition 3,4 had come up, before the commit call for partition 0 was reached. Then, the new partition group info would be: [0, 1, 2, 3, 4]

Reply:
For our discussion: What if partition group 0 is completing, sets to DONE, and we find that the latest segment in partitionGroup 1 is in OFFLINE state in idealstate because of some consumption errors (possibly transient)?
Maybe this is handled, just wanted to add it here so we don't forget about it

Reply:
The `getCurrentPartitionGroupStatus` call will not look at the state in ideal state. So even if it is OFFLINE, it will be included in the partitionGroupStatus, with Status=whatever's in the ZK metadata. So it would still be
0 - seqNum: 0, startOffset: 100, endOffset: 220, status: DONE
1 - seqNum: 0, startOffset: 110, endOffset: -, status: IN_PROGRESS